packages/brisa-eslint-plugin/index.ts

```diff
@@ -0,0 +1,70 @@
+import { ESLintUtils, type TSESTree } from '@typescript-eslint/utils';
+
+export const rules = {
+  'no-server-event-prevent': ESLintUtils.RuleCreator(
+    (name) => `https://eslint.org/docs/rules/${name}`,
+  )<[], 'preventUsage'>({
+    name: 'no-server-event-prevent',
+    meta: {
+      type: 'problem',
+      docs: {
+        description:
+          'Disallow preventDefault() and stopPropagation() on server events',
+      },
+      schema: [],
+      messages: {
+        preventUsage: 'Avoid using {{ method }} on server events.',
+      },
+    },
+    defaultOptions: [],
+    create(context) {
+      const SERVER_EVENTS = new Set([
+        'load',
+        'DOMContentLoaded',
+        'beforeunload',
+        'unload',
+      ]);
+
+      return {
+        CallExpression(node: TSESTree.CallExpression) {
+          if (
+            node.callee.type === 'MemberExpression' &&
+            node.callee.object.type === 'Identifier' &&
+            node.callee.property.type === 'Identifier' &&
+            (node.callee.property.name === 'preventDefault' ||
+              node.callee.property.name === 'stopPropagation')
+          ) {
+            const parent = node.parent;
+            if (
+              parent &&
+              parent.type === 'CallExpression' &&
+              parent.callee.type === 'MemberExpression' &&
+              parent.callee.object.type === 'Identifier' &&
+              parent.callee.property.type === 'Identifier' &&
+              parent.callee.property.name === 'addEventListener'
```

Comment:
The idea of this rule is not using `addEventListener` because you can't register events on the server, but inside the JSX.

That is to say, this is not possible because the window does not exist in the server components:

```js
window.addEventListener('load', someEvent);
```

Server Actions are registered like this:

```tsx
<button onClick={someServerAction} />
```